server/src/Controller/User/UserController.js

```diff
@@ -0,0 +1,80 @@
+import Repository from '../../Repositories/genericRepository';
+import successHandler from '../../libs/routes/successHandler';
+import { userModel } from '../../Model';
+
+const repository = new Repository(userModel);
+
+const getUser = async (req, res, next) => {
+  try {
+    const { id } = req.params;
+    const result = await repository.findOne(id);
+    if (!result) {
+      const err = new Error('User does not exists');
+      next(err);
+    }
+    res
+      .status(200)
+      .send(successHandler('User is here', 200, result));
+  } catch (error) {
+    next(error);
+  }
+};
+
+const getAllUsers = async (req, res, next) => {
+  try {
+    const result = await repository.findMany();
+    res
+      .status(200)
+      .send(successHandler('Users are here', 200, result));
+  } catch (error) {
+    next(error);
+  }
+};
+
+const createUser = async (req, res, next) => {
+  try {
+    const { name, email, password } = req.body;
+    const data = { name, email, password };
+    const result = await repository.create(data);
+
+    res
+      .status(200)
+      .send(successHandler('User created successfully', 200, result));
+  } catch (error) {
+    next(error);
+  }
+};
+
+const deleteUser = async (req, res, next) => {
+  try {
+    const { id } = req.params;
+    const result = await repository.delete(id);
+
+    res
+      .status(200)
+      .send(successHandler('User deleted successfully', 200, result));
+  } catch (error) {
+    next(error);
+  }
+};
+
+const updateUser = async (req, res, next) => {
+  try {
+    const { id, dataToUpdate } = req.body;
+    const result = await repository.update(id, dataToUpdate);
+
+    res
+      .status(200)
```

Comment:
Why setting status here, when you are doing same thing in succrsshandler?

Reply:
I have just removed it from the 'successHandler'.